src/frontend/js/components/SaleTunnel/AddressSelector/CreateAddressFormModal.tsx

```diff
@@ -0,0 +1,66 @@
+import { Button, Modal, ModalProps, ModalSize } from '@openfun/cunningham-react';
+import { defineMessages, FormattedMessage, useIntl } from 'react-intl';
+import { Address } from 'types/Joanie';
+import { AddressFormValues, useDashboardAddressForm } from 'hooks/useDashboardAddressForm';
+import { useAddressesManagement } from 'hooks/useAddressesManagement';
+import { Spinner } from 'components/Spinner';
+import Banner, { BannerType } from 'components/Banner';
+
+const messages = defineMessages({
+  title: {
+    defaultMessage: 'Add address',
+    description: 'Add address modal title.',
+    id: 'components.SaleTunnel.AddressSelector.CreateAddressFormModal.title',
+  },
+  submit: {
+    defaultMessage: 'Submit',
+    description: 'Submit button label.',
+    id: 'components.SaleTunnel.AddressSelector.CreateAddressFormModal.submit',
+  },
+});
+
+export interface AddressFormModalProps extends Pick<ModalProps, 'isOpen' | 'onClose'> {
+  onSettled?: (address: Address) => void;
```

Comment:
first time i see this wording, i'm more use to "onSave", "onCreate" or "onSuccess".

Reply:
Yep I named it this way because it is used as the same time by the create and edit modal, it is agnostic

Reply:
Example: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled